test/Test/Time/Property.hs

```diff
@@ -57,45 +58,47 @@ verifyAnyTime (MkAnyTime t) = read (show t) === t
 
 -- | Verifier for 'toUnit'.
 verifyToUnit :: forall m . (MonadTest m) => AnyTime -> AnyTime -> m ()
-verifyToUnit (MkAnyTime t1) (MkAnyTime t2) = asd t1 t2
+verifyToUnit (MkAnyTime t1) (MkAnyTime t2) = checkToUnit t1 t2
   where
-    asd :: forall (a :: Rat) (b :: Rat) .
-           (KnownSymbol (UnitName a), KnownRat a, KnownRat b)
-        => Time a
-        -> Time b
-        -> m ()
-    asd t _ = withRuntimeDivRat @b @a (toUnit @a (withRuntimeDivRat @a @b $ toUnit @b t)) === t
+    checkToUnit :: forall (unitFrom :: Rat) (unitTo :: Rat) .
+                   (KnownSymbol (UnitName unitFrom), KnownRat unitFrom, KnownRat unitTo)
+                => Time unitFrom
+                -> Time unitTo
+                -> m ()
+    checkToUnit t _ = withRuntimeDivRat @unitTo @unitFrom
+                    $ withRuntimeDivRat @unitFrom @unitTo
+                    $ toUnit (toUnit @unitTo t) === t
 
 -- | Generates random number from [0, 8].
-getNumber0'9 :: (MonadGen m) => m Int
-getNumber0'9 = Gen.int $ Range.constant 0 9
+number0'9 :: (MonadGen m) => m Int
+number0'9 = Gen.int $ Range.constant 0 9
 
 -- | Generates random natural number up to 10^20.
 -- it receives the lower bound so that it wouldn't be possible
 -- to get 0 for denominator.
-getNatural :: (MonadGen m) => Natural -> m Natural
-getNatural n = Gen.integral (Range.constant n $ 10 ^ (20 :: Int))
+natural :: (MonadGen m) => Natural -> m Natural
+natural n = Gen.integral (Range.constant n $ 10 ^ (20 :: Int))
 
 -- | Generates random rational number.
-getRationalNum :: (MonadGen m) => m RatioNat
-getRationalNum = do
-    numeratorVal <- getNatural 0
+rationalNum :: (MonadGen m) => m RatioNat
+rationalNum = do
+    numeratorVal <- natural 0
     isOne        <- Gen.bool
     denomVal     <- if isOne then pure 1
-                             else getNatural 1
+                             else natural 1
     return $ numeratorVal % denomVal
 
+numRatPair :: (Monad m) => PropertyT m (Int, RatioNat)
+numRatPair = liftA2 (,) (forAll number0'9) (forAll rationalNum)
```

Comment:
You always call `unitChooser` function on the both arguments of pair. So it's even better to call `unitChooser` directly here and make this function have name `anyTime` and return `PropertyT m AnyTime`.